File: programming files/rgb_to_hex.java
public class rgb_to_hex {
    public static void main(String[] args) {
        // sets r g b values
        int r = 255;
        int g = 127;
        int b = 0;
        String hexColor = rgbToHex(r, g, b); // calls rgbToHex function
        System.out.println("RGB color (" + r + ", " + g + ", " + b + ") = " + hexColor); // prints out the hex value
    }

    public static String rgbToHex(int r, int g, int b) {
        // ensures that the values are in the range of values
        r = Math.min(255, Math.max(0, r));
        g = Math.min(255, Math.max(0, g));
        b = Math.min(255, Math.max(0, b));
        return String.format("%02X%02X%02X", r, g, b); // returns the hex value
    }
}

//Test with RGB color (255, 127, 0) = FF7F00

File: programming files/rgb_to_hex.cc
#include <iostream>
#include <sstream>
#include <iomanip>
// takes in three r g b values
std::string rgb_to_hex(int r, int g, int b)
{
    // makes sure the number is between 0 and 255 for each of the rgb values
    r = std::max(0, std::min(255, r));
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));

    std::stringstream ss;
    // converts the numbers into hexadecimal form
    ss << std::uppercase << std::hex << std::setfill('0')
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;

    return ss.str();
}

//Test with std::string hexColor = rgb_to_hex(255, 127, 0); // returns "FF7F00"
